File: app/src/main/java/com/codylimber/fieldphenology/ui/screens/specieslist/MiniBarChart.kt
package com.codylimber.fieldphenology.ui.screens.specieslist

import androidx.compose.foundation.Canvas
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import com.codylimber.fieldphenology.data.model.WeeklyEntry
import com.codylimber.fieldphenology.ui.theme.Primary

@Composable
fun MiniBarChart(
    weekly: List<WeeklyEntry>,
    currentWeek: Int,
    modifier: Modifier = Modifier,
    barColor: Color = Primary
) {
    Canvas(modifier = modifier) {
        val w = size.width
        val h = size.height
        val barWidth = w / 53f
        val gap = barWidth * 0.15f

        // Draw bars
        for (entry in weekly) {
            if (entry.week < 1 || entry.week > 53) continue
            val x = (entry.week - 1) * barWidth
            val barH = entry.relAbundance * h
            if (barH > 0.5f) {
                val alpha = 0.3f + 0.7f * entry.relAbundance
                drawRect(
                    color = barColor.copy(alpha = alpha),
                    topLeft = Offset(x + gap / 2, h - barH),
                    size = Size(barWidth - gap, barH)
                )
            }
        }

        // Current week indicator
        if (currentWeek in 1..53) {
            val cx = (currentWeek - 0.5f) * barWidth
            drawLine(
                color = Color.White.copy(alpha = 0.7f),
                start = Offset(cx, 0f),
                end = Offset(cx, h),
                strokeWidth = 1.5f
            )
        }
    }
}

File: ios/Manakin/UI/Components/MiniBarChart.swift
import SwiftUI

struct MiniBarChart: View {
    let weekly: [WeeklyEntry]
    let currentWeek: Int
    var barColor: Color = .appPrimary

    var body: some View {
        Canvas { context, size in
            let w = size.width
            let h = size.height
            let barWidth = w / 53.0
            let gap = barWidth * 0.15

            // Draw bars
            for entry in weekly {
                guard entry.week >= 1, entry.week <= 53 else { continue }
                let x = CGFloat(entry.week - 1) * barWidth
                let barH = CGFloat(entry.relAbundance) * h
                guard barH > 0.5 else { continue }
                let alpha = 0.3 + 0.7 * Double(entry.relAbundance)
                let rect = CGRect(
                    x: x + gap / 2,
                    y: h - barH,
                    width: barWidth - gap,
                    height: barH
                )
                context.fill(
                    Path(rect),
                    with: .color(barColor.opacity(alpha))
                )
            }

            // Current week indicator
            if currentWeek >= 1, currentWeek <= 53 {
                let cx = (CGFloat(currentWeek) - 0.5) * barWidth
                var path = Path()
                path.move(to: CGPoint(x: cx, y: 0))
                path.addLine(to: CGPoint(x: cx, y: h))
                context.stroke(
                    path,
                    with: .color(.white.opacity(0.7)),
                    lineWidth: 1.5
                )
            }
        }
        .frame(height: 20)
    }
}
